src/Ship.js

```diff
@@ -0,0 +1,41 @@
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary
+    this.currentPort = itinerary.ports[0]
+    this.currentPort.addShip(this)
+    this.previousPort = null
+  }
+  setSail() {
+    if (!this.currentPort) {
+      throw new Error("The ship is not docked")
+    }
+
+    const hasNextPort =
+      this.itinerary.ports.indexOf(this.currentPort) + 1 <
+      this.itinerary.ports.length
+
+    if (!hasNextPort) {
+      throw new Error("The ship has arrived at its final destination")
+    }
+
+    this.currentPort.removeShip(this)
+    this.previousPort = this.currentPort
+    this.currentPort = null
+  }
+  dock() {
+    if (this.currentPort) {
+      throw new Error("Ship is already docked")
+    }
+
+    const indexOfPreviousPort = this.itinerary.ports.indexOf(this.previousPort)
+    const indexOfNextPort = indexOfPreviousPort + 1
+    const hasNextPort = indexOfNextPort < this.itinerary.ports.length
+
+    this.currentPort = this.itinerary.ports.at(indexOfNextPort)
```

Comment:
This line is causing the tests to fail. You need to assign the current port the value of the next port which looks to be what you are going for, just the syntax is wrong:
```
    this.currentPort = this.itinerary.ports[indexOfPreviousPort + 1]
```
Doing it like this means you can also delete lines 31 & 32 (line 32 isnt being used anywhere)